torch/csrc/inductor/aoti_runtime/model_container.h

```diff
@@ -299,6 +284,50 @@ class AOTInductorModelContainer {
     lk.lock();
     available_models_.push_back(model);
   }
+
+  uint8_t* constant_ptr(bool is_cpu, size_t constant_offset,
+      size_t bytes_read, size_t data_size) {
+    uint8_t* internal_ptr;
+    if (is_cpu) {
+      // get pointer to constant which is packed in model during compile time.
+      internal_ptr =
+          const_cast<uint8_t*>(_binary_constants_bin_start) + bytes_read;
+    } else {
+#ifdef USE_CUDA
+      auto* constants_ptr = static_cast<uint8_t*>(constant_blob_.get());
+      internal_ptr = constants_ptr + constant_offset;
+      // Copy data to GPU memory
+      // TODO: Handle shared storage case.
+      AOTI_RUNTIME_DEVICE_CHECK(cudaMemcpy(
+          internal_ptr,
+          _binary_constants_bin_start + bytes_read,
+          data_size,
+          cudaMemcpyHostToDevice));
+#else // !USE_CUDA
```

Comment:
Probably don't need the else branch here? My general feeling is `USE_CUDA` and `is_cpu` start to make the code fragmented. Some of the code can be cleaned up a bit, but it seems like using a macro like `USE_CUDA` is inevitable.

Reply:
Yeah, I have the same concern. I think ```USE_CUDA``` is inevitable. On the other hand, I think ```is_cpu``` seems to be useful for cases where we had mix-arch execution. I will try to minimize the use of ```is_cpu``` for now and refine it later when we have a better design for supporting multiple arches.